90zfsbootmenu/zfsbootmenu-lib.sh

```diff
@@ -285,14 +285,20 @@ select_kernel() {
 find_kernel_args() {
   local zfsbe
   zfsbe="${1}"
-  local selected_args
+  local selected_args zfsbe_args
 
   if [ -f "${BASE}/default_args" ]
   then
     cat "${BASE}/default_args"
     return
   fi
 
+  zfsbe_args="$( zfs get -H -o value org.zfsbootmenu:commandline ${zfsbe} )"
```

Comment:
`${zfsbe}` is perhaps badly named, in this case it's the full path to where the boot environment is mounted. An example call is at https://github.com/zdykstra/zfsbootmenu/blob/master/90zfsbootmenu/zfsbootmenu-lib.sh#L249

`find_kernel_args()` should probably accept a second, optional parameter that's the name of the ZFS file system, like `find_kernel()` takes.

Reply:
OK, the latest push addresses this with the following logic:

1. The BE dataset name is now the first argument and the mountpoint is the second argument; if the property is the preferred source for the arguments, its requirement should come first.

2. The property is only checked if the dataset argument is non-empty.

3. The `etc/default/grub` file is only checked if the mountpoint argument is non-empty.

This should robustly handle all sorts of issues discovering arguments, with the Right Thing (TM) done if the function is provided with both the dataset name and mountpoint. Both calls to this function have been updated to provide the expected arguments.